src/Sylius/Bundle/MailerBundle/Sender/Adapter/SwiftMailerAdapter.php

```diff
@@ -57,6 +58,12 @@ public function send(
 
         $message->setBody($renderedEmail->getBody(), 'text/html');
 
+        foreach($attachments as $attachment) {
+            $file = \Swift_Attachment::fromPath($attachment);
+
+            $message->attach($file);
```

Comment:
Why not just `$massage->attach(\Swift_Attachment::fromPath($attachment));`?

Reply:
@Zales0123 , @pjedrzejewski , @pamil 
I had an issue, where I was using S3 pre-signed URLs and my file name was showing up with all the extra amazon signature etc from query parameters from the signed URL and Gmail broke my pdf preview as it couldn't recognize the file. 
How can I overwrite the Swift attachment and set the filename, content type, mime type? I know we can use this?
```\Swift_Attachment::fromPath("<S3 PATH>")->setDisposition('inline')->setFilename("<FILENAME>")->setContentType("application/pdf")```


If I send the \SwiftAttachment objects in attachment array by customizing it, this code will override it, how can I overcome that?
```$file = \Swift_Attachment::fromPath($attachment);```